### PR TITLE
Add the offset and size to the dump SST footer details

### LINKED ISSUE #404: sst_dump tool should show both the offset and size from the index block handle and metaindex block handle
Owner: 

**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by git-hulk**
Before applying this PR, the footer details:

```
Footer Details:
--------------------------------------
  metaindex handle: B0E499405C
  index handle: 8AC49940CD17
  table_magic_number: 9863518390377041911
  format version: 5
```

and after

```
Footer Details:
--------------------------------------
  metaindex handle: B0E499405C offset: 134640176 size: 92
  index handle: 8AC49940CD17 offset: 134636042 size: 3021
  table_magic_number: 9863518390377041911
  format version: 5
```

This closes #404

### CODE REVIEW

File: table/format.cc [hunk 1]
```diff
@@ -328,17 +328,15 @@ std::string Footer::ToString() const {
...
-                  "\n  ");
+  result.append("metaindex handle: " + metaindex_handle_.ToString() +
+                " offset: " + std::to_string(metaindex_handle_.offset()) +
+                " size: " + std::to_string(metaindex_handle_.size()) + "\n  ");
+  result.append("index handle: " + index_handle_.ToString() +
+                " offset: " + std::to_string(index_handle_.offset()) +
+                " size: " + std::to_string(index_handle_.size()) + "\n  ");
+  result.append("table_magic_number: " + std::to_string(table_magic_number_) +
+      "\n  ");
+  if (!IsLegacyFooterFormat(table_magic_number_)) {
```
Comment (git-hulk) [inline table/format.cc:339]: I moved the first three lines out since they both appeared in both of if and else blocks.

**Comment by git-hulk** [comment]
@udi-speedb Thanks for your review. I have updated the HISTORY.md, please take a look.

**Comment by udi-speedb** [comment]
@git-hulk Thanks for that.

Just one last minor thing . Please add the issue's number to the end of the description you have added to the HISTORY.md like so:
`* sst_dump: display metaindex_handle and the index_handle's offset and size in footer information (#404).`

**Comment by git-hulk** [comment]
> @git-hulk Thanks for that.
> 
> Just one last minor thing . Please add the issue's number to the end of the description you have added to the HISTORY.md like so: `* sst_dump: display metaindex_handle and the index_handle's offset and size in footer information (#404).`

Sure, sorry for missing that.

**Comment by udi-speedb** [comment]
> > @git-hulk Thanks for that.
> > Just one last minor thing . Please add the issue's number to the end of the description you have added to the HISTORY.md like so: `* sst_dump: display metaindex_handle and the index_handle's offset and size in footer information (#404).`
> 
> Sure, sorry for missing that.

No problem. Thanks a lot.

**Comment by udi-speedb** [comment]
@git-hulk 
Please squash your commits into a single one and force push. 
The commit message of the first commit is fine as the commit message for the single, squashed commit. Just please also add the issue number at the end of that commit message:
`Add the offset and size into the dump SST footer detail (#404)`

Once done, I will approve the PR.

**Comment by git-hulk** [comment]
@udi-speedb Done, thank you! I will take care of those rules next time.

**Review by udi-speedb — APPROVED** [review]

**Comment by udi-speedb** [comment]
> @udi-speedb Done, thank you! I will take care of those rules next time.

@git-hulk Thanks a lot for your contribution.

**Comment by udi-speedb** [comment]
@git-hulk I apologize, but I forgot to ask you to run `make format` on your code. So, please do and force push again.

**Comment by git-hulk** [comment]
Looks like I shall add the Apache 2.0 header for this modified file?

**Comment by git-hulk** [comment]
> @git-hulk I apologize, but I forgot to ask you to run `make format` on your code. So, please do and force push again.

My bad, I should go through CONTRIBUTED.md doc first before writing code. Thanks for your guidance.

**Review by udi-speedb — APPROVED** [review]

**Comment by udi-speedb** [comment]
> > @git-hulk I apologize, but I forgot to ask you to run `make format` on your code. So, please do and force push again.
> 
> My bad, I should go through the CONTRIBUTING.md doc first before writing the code. Thanks for your guidance.

No problem. Thanks again for your cooperation and contribution.

**Comment by git-hulk** [comment]
@udi-speedb Looks like the check history and license are not correct, should I fix it?

https://github.com/speedb-io/speedb/blob/main/.github/workflows/check_license_and_history.yml#L36

I guess it should checkout the repository before checking.

**Comment by udi-speedb** [comment]
> @udi-speedb Looks like the check history and license are not correct, should I fix it?
> 
> https://github.com/speedb-io/speedb/blob/main/.github/workflows/check_license_and_history.yml#L36
> 
> I guess it should checkout the repository before checking.

No need. It's a known issue. We need to resolve it on our side. Thanks

**Comment by git-hulk** [comment]
@udi-speedb Should I worry about the failure of the test? It looks like didn't relate to PR's change.

**Comment by udi-speedb** [comment]
> @udi-speedb Should I worry about the failure of the test? It looks like didn't relate to PR's change.

Indeed seems totally unrelated to your change. I am handling this. Will let you know if anything else is needed. Thanks

**Review by udi-speedb — APPROVED** [review]

**Review by udi-speedb — APPROVED** [review]

**Review by udi-speedb — APPROVED** [review]

**Comment by udi-speedb** [comment]
@git-hulk I aplogize, but I need to ask you to update the license text at the top of the file you have modified. We need to update our CONTRIBUTING.md file to reflect this recent decision. You may find the text to add in this issue:
#713 .
As an example, you may see column_family.cc

Please do not force push your updates. Just add commits. At the time of merging, we will squash and leave only the applicable commit message.
Thanks for your cooperation.

**Review by udi-speedb — APPROVED** [review]

**Comment by git-hulk** [comment]
Hi @udi-speedb It's my pleasure, thanks for you review.

**Pull request merged**